### DISCUSSION TITLE
Branch protection

### DISCUSSION THREAD

**Discussion by rettigl** [OpenCOMPES/sed#285] [General]
It appears branch protection for main has been disabled. Did any of you do this? If so, why?

**Comment by rettigl** [comment]
@zain-sohail did you change this for allowing the publish workflow to bump the version? If so, we should find the solution that allows to keep branch protection on.
Maybe this helps
https://github.com/orgs/community/discussions/25305

**Reply by zain-sohail** [reply]
Yes I did. But the branch is still protected (added back other rules). I intended to just turned off this rule: 
```
Require a pull request before merging
When enabled, all commits must be made to a non-protected branch and submitted via a pull request before they can be merged into a branch that matches this rule
```
You can look here at our protection rules:
https://github.com/OpenCOMPES/sed/settings/branch_protection_rules/43987994

**Reply by rettigl** [reply]
When I wrote this, everything was disabled there. I am open for discussion to change this, but I would find it quite important that we do discuss these kind of changes before applying them.
The current settings don't enforce a pull request, nor an approving review, thus effectively removing branch protection. I don't find this a good solution.

**Reply by zain-sohail** [reply]
yes my bad. I actually just turned it off to test if the push works as expected and then forgot. 
one option i can see is we push to a release branch. For example, for 0.1.6, we already have the prereleases. And those can be pushed there and a pull request created when we finally release 0.1.6. This could maybe mean a slight change for developers to open PRs against the release branch and not main.

**Comment by zain-sohail** [comment]
> @zain-sohail did you change this for allowing the publish workflow to bump the version? If so, we should find the solution that allows to keep branch protection on. Maybe this helps https://github.com/orgs/community/discussions/25305

Regarding this, indeed I have looked into it but github doesn't allow actions to bypass these rules. Only a user or app can. I am not sure how to create an app for this.
Other thing I have seen possible is to create rulesets (new feature) and we can exempt certain actors from the rules

**Reply by rettigl** [reply]
With the current settings, the publish action cannot bump the version, see https://github.com/OpenCOMPES/sed/actions/runs/6942029899
In the link I posted, there is this: https://github.com/CasperWA/push-protected

**Reply by zain-sohail** [reply]
Yes that's only possible through use of a PAT, which is user specific so not recommended. But we can use our PR app token for pushing as well. I put a new PR #291  for such a case.

**Reply by rettigl** [reply]
We can test that out, let's see.

**Comment by zain-sohail** [comment]
Resolved now
